xarray/tests/test_plugins.py

```diff
@@ -1,10 +1,19 @@
 from unittest import mock
 
-import pkg_resources
 import pytest
 
 from xarray.backends import common, plugins
 
+try:
+    import importlib.metadata as importlib_metadata
+
+    from importlib_metadata import EntryPoint
+
+except ImportError:
+    # if the fallback library is missing, we are doomed.
+    import importlib_metadata as importlib_metadata  # type: ignore[no-redef]
+    from importlib_metadata import EntryPoint  # type: ignore[no-redef]
```

Comment:
This can go outside the try-except block. Since it is the same import. Also, shouldn't we add `importlib_metadata` to `ci/requirements` since it is mandatory for tests now and to avoid confusion in the future when tests will be run on Python 3.8+?

Reply:
`isort` will put this above the "try-except block, so this will import from the wrong module.

Reply:
> Also, shouldn't we add `importlib_metadata` to `ci/requirements` since it is mandatory for tests now and to avoid confusion in the future when tests will be run on Python 3.8+?

Not sure if this is needed, since `importlib_metadata` is only the fallback version and `importlib.metadata` comes with Python.

Reply:
> `isort` will put this above the "try-except" block, so this will import from the wrong module.

Have a look at special comments which isort understands: https://pycqa.github.io/isort/docs/configuration/action_comments.html

Reply:
If I understood correctly `importlib_metadata` will only be installed for Python < 3.8.

https://github.com/pydata/xarray/blob/884f0cb20f5dccc346ae1a2114e25839de6bc57e/setup.cfg#L81

And you are mocking only `importlib_metadata.EntryPoint` below.